### PR TITLE
Initial Work

### PR DESCRIPTION
**Pull request by ajschmidt8**
This PR adds the Dockerfile and associated GitHub Actions (for PRs & merges) for the new `rapidsai/mambaforge-cuda` images.

The images simply copy `/opt/conda` from the official [condaforge/mambaforge](https://hub.docker.com/r/condaforge/mambaforge) images and install the expected Python version.

For posterity, the last few lines in the Dockerfile came from inspecting the `mambaforge` Dockerfile [here](https://github.com/conda-forge/miniforge-images/blob/e994b96c3886e1514666360b499ddc940ad87598/ubuntu/Dockerfile):

```dockerfile
# in Dockerfile
RUN \
  # ensure conda environment is always activated
  ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh; \
  echo ". /opt/conda/etc/profile.d/conda.sh; conda activate base" >> /etc/skel/.bashrc; \
  echo ". /opt/conda/etc/profile.d/conda.sh; conda activate base" >> ~/.bashrc; \
  # install expected Python version
  mamba install -y python="${PY_VER}"; \
  find /opt/conda -follow -type f -name '*.a' -delete; \
  find /opt/conda -follow -type f -name '*.pyc' -delete; \
  conda clean -afy

```

### CODE REVIEW

File: .github/workflows/build-and-publish-images.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: Reusable workflow example
```
Comment (raydouglass) [inline .github/workflows/build-and-publish-images.yml:1]: Need to update the name here

**Review by raydouglass — APPROVED** [review]

**Review by raydouglass — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ENV PATH=/opt/conda/bin:$PATH
+
+COPY --from=condaforge/mambaforge:4.12.0-0 /opt/conda /opt/conda
+RUN \
+  # ensure conda environment is always activated
+  ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh; \
+  echo ". /opt/conda/etc/profile.d/conda.sh; conda activate base" >> /etc/skel/.bashrc; \
+  echo ". /opt/conda/etc/profile.d/conda.sh; conda activate base" >> ~/.bashrc; \
+  # install expected Python version
+  mamba install -y python="${PY_VER}"; \
```
Comment (jakirkham) [inline Dockerfile:16]: Might want to do an update after this for good measure

```suggestion
  mamba install -y python="${PY_VER}"; \
  mamba update --all -y; \
```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  find /opt/conda -follow -type f -name '*.a' -delete; \
+  find /opt/conda -follow -type f -name '*.pyc' -delete; \
```
Comment (jakirkham) [inline Dockerfile:18]: Curious why are we doing this?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  conda clean -afy
```
Comment (jakirkham) [inline Dockerfile:19]: Are we sure we want `-f` here?

File: Dockerfile (hunk 3 above)
Comment (ajschmidt8) [reply Dockerfile:18]: @jakirkham, I found these lines from the miniforge Dockerfile below. In the comments they mention an article that discusses conda/Docker clean-up tips

- https://github.com/conda-forge/miniforge-images/blob/e994b96c3886e1514666360b499ddc940ad87598/ubuntu/Dockerfile#L35-L37

File: Dockerfile (hunk 4 above)
Comment (ajschmidt8) [reply Dockerfile:19]: miniforge does it. do you see any issues with it?

- https://github.com/conda-forge/miniforge-images/blob/e994b96c3886e1514666360b499ddc940ad87598/ubuntu/Dockerfile#L38

File: Dockerfile (hunk 2 above)
Comment (ajschmidt8) [reply Dockerfile:16]: good point. i will open a PR for this.

Comment (ajschmidt8) [reply Dockerfile:16]: @jakirkham, PR here https://github.com/rapidsai/mambaforge-cuda/pull/3

**Pull request merged**